core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java

```diff
@@ -53,12 +60,26 @@ public ReflectiveClassBuildItem(boolean constructors, boolean methods, boolean f
         this(constructors, methods, fields, false, false, className);
     }
 
+    public ReflectiveClassBuildItem(boolean constructors, boolean methods, boolean fields, boolean serialization,
+            String... className) {
+        this(constructors, methods, fields, false, false, serialization, className);
+    }
+
     public static ReflectiveClassBuildItem weakClass(String... className) {
         return new ReflectiveClassBuildItem(true, true, true, false, true, className);
     }
 
+    public static ReflectiveClassBuildItem serializationClass(String... className) {
+        return new ReflectiveClassBuildItem(false, false, false, false, false, true, className);
+    }
+
     private ReflectiveClassBuildItem(boolean constructors, boolean methods, boolean fields, boolean finalFieldsWritable,
             boolean weak, String... className) {
+        this(constructors, methods, fields, false, false, false, className);
```

Comment:
This is wrong, it needs to pass in the values on finalFieldsWritable and weak, not just false. This is why there are test failures.

Reply:
Thank you for spotting that. I'm not sure why I changed the original behavior (seems like a copy/paste problem). It is fixed now.